models/dataset.py

```diff
@@ -53,6 +53,37 @@ def end_date(self) -> datetime:
 
         return time_values.max()
 
+    @cached_property
+    def date_cutoff(self) -> datetime:
+        # The cut-off date for dataset train/test split
+        df = self.df
+        dates_unique = df[self.time_column].unique()
+        date_cutoff = sorted(dates_unique)[
+            round(len(dates_unique) * self.train_percentage)
+        ]
+
+        return date_cutoff
+
+    @property
+    def df_train(self) -> pd.DataFrame:
+        df = self.df
+        df[self.time_column] = df[self.time_column].astype("datetime64[ns]")
```

Comment:
Maybe just do this in the original `def df` method.

Reply:
As mentioned in: https://github.com/ivanmkc/forecasting-live-demo-backend/pull/4/files/3b3ab0068739264763ca7e2265bef78760a7de03#r1028595898